internal/cliwrapper/cliwrapper.go

```diff
@@ -93,22 +81,45 @@ func (p *cliWrapper) Deploy(image string, podmanArgs []string, containerArgs []s
 }
 
 func (p *cliWrapper) KillAndClean(containerName string) error {
-	cmdKill := exec.Command(p.podmanFullPath, "kill", containerName) //nolint:gosec
+	cmdKill := p.getPodmanCmd("kill", containerName)
 	p.logger.Debugf("Killing with command %v", cmdKill.Args)
 	if err := cmdKill.Run(); err != nil {
 		p.logger.Warningf("failed to kill pod %s, probably the execution terminated earlier", containerName)
 	} else {
 		p.logger.Warningf("successfully killed container %s", containerName)
 	}
 
-	var cmdRmContainerStderr bytes.Buffer
-	cmdRmContainer := exec.Command(p.podmanFullPath, "rm", "--force", containerName) //nolint:gosec
-	p.logger.Debugf("Removing container with command %v", cmdRmContainer.Args)
-	cmdRmContainer.Stderr = &cmdRmContainerStderr
-	if err := cmdRmContainer.Run(); err != nil {
-		p.logger.Errorf("failed to remove container %s: %s", containerName, cmdRmContainerStderr.String())
+	msg := "removing container " + containerName
+	_, err := p.runPodmanCmd(msg, "rm", "--force", containerName)
+	if err != nil {
+		p.logger.Errorf(err.Error())
 	} else {
 		p.logger.Infof("successfully removed container %s", containerName)
 	}
 	return nil
 }
+
+func (p *cliWrapper) getPodmanCmd(cmdArgs ...string) *exec.Cmd {
+	var commandArgs []string
+	if len(p.runtimeContext) > 0 && p.runtimeContext[0] != "" {
+		commandArgs = append(commandArgs, p.runtimeContext...)
+	}
```

Comment:
It would be better if this code only had to check one thing instead of two.  For instance, if `NewCliWrapper()` ensured that `p.runtimeContext` contained _no entries_ when there were no options to provide, then we could reduce this code to just:
```go
	commandArgs := p.runtimeContext
```
Because `p.runtimeContext` would either be an empty list or a list with the stuff that we already want -- either way, we wouldn't need the conditional.

Reply:
Made the above suggested changes and added the check here:
https://github.com/arcalot/arcaflow-engine-deployer-podman/pull/61/commits/eb6cd6347bb2e27d49d5797138c2569f9910aa78#diff-1123427b979173bab05da57bb98e3033f246de79c9d274320e84fd4f3b22e68bR24-R25